lib/msf/util/db_manager.rb

```diff
@@ -1,19 +1,41 @@
 module Msf
-  module Util
-    module DBManager
-      # Creates search conditions to match the specified search string against all of the model's columns.
-      #
-      # @param model - An ActiveRecord model object
-      # @param search - A string regex search
-      # @return Arel::Nodes::Or object that represents a search of all of the model's columns
-      def self.create_all_column_search_conditions(model, search)
-        search = "(?mi)#{search}"
-        condition_set = model.columns.map do |column|
-          Arel::Nodes::Regexp.new(Arel::Nodes::NamedFunction.new("CAST", [model.arel_table[column.name].as("TEXT")]),
-                                  Arel::Nodes.build_quoted(search))
-        end
-        condition_set.reduce { |conditions, condition| conditions.or(condition).expr }
-      end
+module Util
+module DBManager
+  # Creates search conditions to match the specified search string against all of the model's columns.
+  #
+  # @param model - An ActiveRecord model object
+  # @param search - A string regex search
+  # @param column_name_skip_list - An array of strings containing column names to skip
+  # @return Arel::Nodes::Or object that represents a search of all of the model's columns
+  def self.create_all_column_search_conditions(model, search, column_name_skip_list=nil)
+    search = "(?mi)#{search}"
+    # remove skip columns
+    columns = model.columns.reject { |column|
+      column_name_skip_list && column_name_skip_list.include?(column.name)
+    }
+
+    condition_set = columns.map { |column|
+      Arel::Nodes::Regexp.new(Arel::Nodes::NamedFunction.new("CAST", [model.arel_table[column.name].as("TEXT")]),
+                              Arel::Nodes.build_quoted(search))
+    }
+    condition_set.reduce { |conditions, condition| conditions.or(condition).expr }
+  end
+
+  def self.process_opts_workspace(opts, framework, required = true)
+    wspace = delete_opts_workspace(opts)
+    if required && (wspace.nil? || ((wspace.kind_of? String) && wspace.empty?))
+      raise ArgumentError.new("opts must include a valid :workspace")
+    end
+
+    if wspace.kind_of? String
+      wspace = framework.db.find_workspace(wspace)
     end
+    wspace
+  end
+
+  def self.delete_opts_workspace(opts)
+    opts.delete(:workspace) || opts.delete(:wspace)
```

Comment:
Should we be removing both `:workspace` and `:wspace`? If both keys are in `opts`, this would return the value for `:workspace`, however, `:wspace` would remain in `opts`. It is definitely an edge case, perhaps it is an exception should `opts` contain both.

Reply:
I agree, we should prefer `:workspace`. Sure, logging a warning would be fine!